### DISCUSSION TITLE
Error(s) trying to inject Appservice into Blazor page.

### DISCUSSION THREAD

**Discussion by mattcole-42** [abpframework/abp#12115] [General] [answered]
I realise this is something I have missed or failed to do, but I can't work it out.

I have 2 modules called Common, and ClubManagement)

I am trying to inject a lookupAppService from the Common Module into the blazor (code behind) form, to populate a drop down with values out of the 'common' module.

If I try the method of: 
```
private readonly ILookupAppService _lookupAppService;
private readonly IClubAppService _clubAppService;

    public Signup(
        IClubAppService clubAppService, 
        ILookupAppService lookupAppService
    )
    {
        _clubAppService = clubAppService;
        _lookupAppService = lookupAppService;
        NewClub = new CreateClubDto();
    }
```

I get this error:

`Unhandled exception rendering component: An exception was thrown while activating R.L.ClubManagement.Blazor.Pages.ClubSignUp.Signup.
Autofac.Core.DependencyResolutionException: An exception was thrown while activating R.L.ClubManagement.Blazor.Pages.ClubSignUp.Signup.
 ---> Autofac.Core.DependencyResolutionException: None of the constructors found with 'Autofac.Core.Activators.Reflection.DefaultConstructorFinder' on type 'R.L.ClubManagement.Blazor.Pages.ClubSignUp.Signup' can be invoked with the available services and parameters:`

I have also tried '[Inject] protected ILookupAppService {get; set;}' and I get the below unregistered service error:

` Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Cannot provide a value for property 'LookupAppService' on type 'R.L.ClubManagement.Blazor.Pages.ClubSignUp.Signup'. There is no registered service of type 'R.L.Common.Lookup.ILookupAppService'.
System.InvalidOperationException: Cannot provide a value for property 'LookupAppService' on type 'R.L.ClubManagement.Blazor.Pages.ClubSignUp.Signup'. There is no registered service of type 'R.L.Common.Lookup.ILookupAppService'.`

I have a second AppService that I have injected the same way and it works, but it is defined in the same module (ClubManagement), so I can only guess that I have failed to do a step to get the cross module AppService(s) working, but I have no idea where or what I have missed.

Thank you in advance for any help.

**Comment by realLiangshiwei** [comment]
Can you provide a project to reproduce?

**Comment by mattcole-42** [comment, marked as answer]
In trying to create a small project to share with the problem replicated, I Discovered what I had done wrong.

I had missed a reference to one of the modules in a DependsOn[] section... (well I had referenced the wrong one)

Thanks for the help anyway.
